samples/shopping-cart-with-view-quickstart/src/main/java/shoppingcart/application/ShoppingCartView.java

```diff
@@ -0,0 +1,103 @@
+package shoppingcart.application;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.annotations.Consume;
+import akka.javasdk.annotations.Query;
+import akka.javasdk.view.TableUpdater;
+import akka.javasdk.view.View;
+import shoppingcart.domain.ShoppingCartEvent;
+
+@ComponentId("shopping_cart_view")
+public class ShoppingCartView extends View {
+
+  @Query("SELECT * FROM shopping_cart_view WHERE cartId = :cartId") // <1>
+  public QueryEffect<Cart> getCart(String cartId) {
+    return queryResult();
+  }
+
+  @Query("SELECT * FROM shopping_cart_view WHERE userId = :userId AND checkedout = false") // <2>
+  public QueryEffect<Cart> getUserCart(String userId) {
```

Comment:
I am not sure how important that is, but technically, you can create many shopping carts for a user, and then this query will fail.